docs: improve docs for `loki.secretfilter` component (#3331)
grafana/alloy: docs/sources/reference/components/loki/loki.secretfilter.md

{{< docs/shared lookup="stability/experimental.md" source="alloy" version="<ALLOY_VERSION>" >}}

`loki.secretfilter` receives log entries and redacts detected secrets from the log lines.

> The detection is based on regular expression patterns, defined in the Gitleaks configuration file embedded within the component.

Reviewer: ```suggestion
The detection relies on regular expression patterns, defined in the Gitleaks configuration file embedded within the component.
```

Is `relies` more accurate here? `based on` suggests that there's an extra bit that happens... that it takes the regex patterns as a starting point and builds on top of that.

Author: For now, it's indeed solely regexes (both for identification and allow-listing). In the future, we might add extra checks like entropy of the detected string. Either way, I think you're right and `relies` is probably more appropriate.

| `types`           | `map(string)`        | Types of secret to look for.                               | All types                        | no       |

The `gitleaks_config` argument is the path to the custom `gitleaks.toml` file.

> The Gitleaks configuration file [embedded in the component][embedded-config] is used if you don't provide the path to a custom configuration file.

Reviewer: ```suggestion
If you don't provide the path to a custom configuration file, the Gitleaks configuration file [embedded in the component][embedded-config] is used.
```

Unsupported fields and values in the configuration file are ignored.
{{< /admonition >}}

{{< admonition type="note" >}}

> The embedded configuration file can change from one version of Alloy to another. It's advised to use an external configuration file to ensure consistency.

Reviewer: ```suggestion
The embedded configuration file may change between Alloy versions.
To ensure consistency, use an external configuration file.
```

Unsupported fields and values in the configuration file are ignored.
{{< /admonition >}}

{{< admonition type="note" >}}

> The embedded configuration file may change between Alloy versions.

Reviewer: ```suggestion
The embedded configuration file may change between {{< param "PRODUCT_NAME" >}} versions.
```
